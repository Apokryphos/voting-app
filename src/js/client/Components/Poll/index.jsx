import React from 'react';
import PropTypes from 'prop-types';

function Poll(props) {
  //  Choices lists are static; just use an index
  let key = 0;
  const choices = props.choices.map(c => <li key={key++}>{c}</li>);

  return (
    <div>
      <h1>{props.question}</h1>
      <ul>{choices}</ul>
    </div>
  );
}

Poll.propTypes = {
  /* eslint react/no-unused-prop-types: 0 */
  id: PropTypes.string.isRequired,
  question: PropTypes.string.isRequired,
  choices: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Poll;
