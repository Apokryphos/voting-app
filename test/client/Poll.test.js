import Poll from 'Components/Poll';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

describe('Poll rendering', () => {
  it('Renders OK', () => {
    const component = ReactTestRenderer.create(<Poll id="59fbcdb370318a44f5f911ea" question="Test Question" choices={['a', 'b', 'c']} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
