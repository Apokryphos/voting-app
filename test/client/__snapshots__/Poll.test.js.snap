// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poll rendering Renders OK 1`] = `
<div>
  <h1>
    Test Question
  </h1>
  <ul>
    <li>
      a
    </li>
    <li>
      b
    </li>
    <li>
      c
    </li>
  </ul>
</div>
`;
